dojo/tools/trustwave_fusion_api/parser.py

```diff
@@ -0,0 +1,100 @@
+import json
+import hashlib
+from datetime import datetime
+from dojo.models import Finding, Endpoint
+
+
+class TrustwaveFusionAPIParser(object):
+    """
+    Import Trustwave Fusion Report from its API in JSON format
+    """
+
+    def get_scan_types(self):
+        return ["Trustwave Fusion API Scan"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return "Trustwave Fusion API Scan"
+
+    def get_description_for_scan_types(self, scan_type):
+        return "Trustwave Fusion API report file can be imported in JSON format"
+
+    # turning a json file to string
+    def parse_json(self, file):
+        data = file.read()
+        try:
+            tree = json.loads(str(data, 'utf-8'))
+        except:
+            tree = json.loads(data)
+
+        return tree
+
+    def get_items(self, tree, test):
+        items = {}
+
+        # iterating through each vulnerability
+        for node in tree['items']:
+            item = get_item(node, test)
+
+            item_key = hashlib.sha256("|".join([
+                item.severity,
+                item.title,
+                item.description
+            ]).encode()).hexdigest()
+
+            if item_key in items:
+                items[item_key].unsaved_endpoints.extend(item.unsaved_endpoints)
+                items[item_key].nb_occurences += 1
+            else:
+                items[item_key] = item
+
+        return list(items.values())
+
+    def get_findings(self, file, test):
+        if file is None:
+            return None
+
+        tree = self.parse_json(file)
+        if tree:
+            return self.get_items(tree, test)
+
+    def convert_severity(self, num_severity):
+        """Convert severity value"""
+        if num_severity >= -10:
+            return "Low"
+        elif -11 >= num_severity > -26:
+            return "Medium"
+        elif num_severity <= -26:
+            return "High"
+        else:
+            return "Info"
+
+
+def get_item(vuln, test):
+    finding = Finding(
+        test=test,
+        unique_id_from_tool=vuln['id'],
+        nb_occurences=1,
+    )
+
+    # Endpoint
+    if vuln['location']['url']:
+        endpoint = Endpoint.from_uri(vuln['location']['url'])
+        finding.saved_endpoints = [endpoint]
+
+    finding.title = vuln['name']
+    finding.description = vuln['classification']
+    finding.severity = f"{vuln['severity'][0]}{vuln['severity'][1:].lower()}"
+
+    # Date
+    date_str = vuln['updatedOn']
+    date_str = date_str[:len(date_str) - 3] + date_str[-2:]
+    finding.date = datetime.strptime(date_str, "%Y-%m-%dT%H:%M:%S.%f%z")
+
+    if "CVE-NO-MATCH" not in vuln['kb']['cves']:
+        cves = ""
+        for cve in vuln['kb']['cves']:
+            cves += f"{cve}, "
+        cves = cves[:len(cves) - 2]  # removing the comma and the blank space
+        finding.cve = cves
```

Comment:
this can't work. the attribute `cve` can hold only one CVE for now :/

Reply:
I think it might be better to include all the cves because the link to the cves will still work: it will link you to the first cve provided.

Reply:
This way if the uploader wants to see other cves, they can still do it manually, and the information would not be lost.

Reply:
Yeah, but you can't because the field is an integer field. You could add the other cves to the description.

Reply:
Can you also add some vulnerabilities/testcases that actually have cves to the logic van be tested/asserted?

Reply:
Sure, will do that.